### PR TITLE
Macbuild

### LINKED ISSUE #77: Versioning
Seems like we need[*] some kind of versioning for releasing exe and app files (#6). There are a couple options:

1. Version number starts off as something like `0.0.1`, and increments automatically every time someone merges a PR to main branch (`0.0.2` after one PR, `0.0.3` after two PRs etc).
2. We manually run a command to increment the version. This seems to be what most projects do, including Porcupine.

With either option, we would configure Github Actions to make a new release with exe and app files when the version number is incremented.

[*] Specifically, we need to tag commits. It means that we say "v0.0.2 means this commit", and then push the tag "v0.0.2" along with the commit. The release, including its exe and app files, will then be associated with the tag.

### PR DESCRIPTION
**Pull request by Akuli**
mac .app file built on github actions for #6

Closes #77

### CODE REVIEW

**Comment by Akuli** [comment]
@ThePhilgrim Seems like `minesweeper.app` is actually a directory, not a file. How should we upload it? Just make a zip file containing all of its contents and ask the users to unzip it first?

**Comment by Akuli** [comment]
Seems like there's also `.dmg` files, whatever they are

**Comment by Akuli** [comment]
@ThePhilgrim Seems like it worked, now try downloading it from releases page: https://github.com/ThePhilgrim/minesweeper/releases

File: .github/workflows/mac_build.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - main
+      - macbuild  # just for testing
+jobs:
+  mac_build:
+    runs-on: macos-11.0
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: "3.9"
```
Comment (Akuli) [inline .github/workflows/mac_build.yml]: Using a brew-installed python might fix the build (https://github.com/actions/setup-python/issues/58):

```suggestion
    - run: brew install python@3.9
```

**Comment by Akuli** [comment]
There are a few problems with this pull request:
- ~Now that #91 got merged, there's also `windows_build.yml`. The yml files should be combined together so that they only make one release, not two separate releases, and the same release contains both `minesweeper.exe` and `minesweeper.dmg`.~ combined
- ~The resulting `Minesweeper.app` just doesn't work. @ThePhilgrim Can you comment here how exactly you tried to run it and paste the errors you got?~ fixed by switching to pyinstaller

**Review by ThePhilgrim — APPROVED** [review]

**Review by Arrinao — APPROVED** [review]

**Pull request merged**